code/game/atom/atom_act.dm

```diff
@@ -75,34 +75,47 @@
 	SEND_SIGNAL(src, COMSIG_ATOM_EMP_ACT, severity, protection)
 	return protection // Pass the protection value collected here upwards
 
+/**
+ * Wrapper for bullet_act used for atom-specific calculations, i.e. armor
+ *
+ * @params
+ * * hitting_projectile - projectile
+ * * def_zone - zone hit
+ * * piercing_hit - is this hit piercing or normal?
+ */
+
+/atom/proc/projectile_hit(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+	return bullet_act(hitting_projectile, def_zone, piercing_hit, check_projectile_armor(def_zone, hitting_projectile))
```

Comment:
shouldn't this still take blocked damage as input to allow for an override of the proc?